src/registry/v1/client.rs

```diff
@@ -44,8 +44,10 @@ where
                 .map_err(|_| ClientError::Request("Failed to get paths".into()))?;
 
             path.extend(&["api", "registry", "v1alpha1", "apps"]);
-            if !application.is_empty() {
-                path.push(application);
+            if let Some(application) = application {
+                if !application.is_empty() {
+                    path.push(application);
+                }
             }
 
             if let Some(device) = device {
```

Comment:
I would append the device in path only if there is an app, to avoid someone calling : 
`url(None, Some("aDevice"))`
Although i realize it's not really necessary as it's not a `pub` function..